### DISCUSSION TITLE
Node configuration

### DISCUSSION THREAD

**Discussion by mzabaluev** [movementlabsxyz/movement#382] [Ideas]
Decide which approach to managing node configuration do we want:

* a single well-defined format (or a serde schema uniformly mapped to JSON, YAML,... )
* a more abstract system with node config, possibly coming from a networked key-value space or similar.

Adjacent issues:

* Requirements from setup components for testing and deployment. Do we need readiness notification as a feature, or can this be managed by setup helpers before the node is started?
* Change notifications in a running node? Current thinking: not worth it, we can always roll over to another instance with the new config in the infrastructure.

**Comment by l-monninger** [comment]
I think a lot of this falls under #384 , but perhaps here we can instigate a pass on all of the configurables in each software unit.

**Comment by l-monninger** [comment]
> Change notifications in a running node 

I also do not think change notifications are clearly needed. However, relying on a notification-based setup task has been easier than before. 

I do think other parts of the current the "manager" pattern are quite nice too. For example, waiting on a valid type contract then having all of the information available to the subsequent units thereafter makes implementations clearer and missing config crashes less likely.
